cfpb/consumerfinance.gov
# Add CFGOV Helm Charts for Local Deployment to K8s

This PR adds a CFGOV Helm chart that allow for deployment to a local K8s cluster. 

## Additions

- A CFGOV Helm Chart that includes:
  - 2 init containers:
    - A Busybox container to ensure that the Postgresql and Opensearch Services are running
    - A data migration container that loads test data into the Postgresql Service and indexes that data to the Opensearch service.
  - 2 containers:
    - The CFGOV application (running on PORT 8000)
    - The Apache server that proxies to to that web application (running on PORT 80)
- Adds `index.sh` and `test.sql.gz` to the CFGOV docker image.
- A helm deployment guide to the master docs
- A `helm-init.sh` script to deploy the helm chart effortlessly.
- A `helm-uninstall.sh` script that tears down everything related to the CFGOV Helm deployment. 
- A Helm notes script that outputs instructions for accessing the CFGOV deployment when deployed locally. 

## How to test this PR

1. Make sure you have a local K8s cluster running.  
2. Clone the PR.   
3. Run: `docker build . -t cfgov`
4. Run: `helm-init.sh`
5. Port forward the cfgov-apache container to a local port.
6. Access your application at `localhost:<local_port>`
7. Ensure the home page is displaying.
8. Visit `localhost:<local_port>/about_us/blog` to make sure the Open Search index is working.

## Notes and todos

- Reach out to @cooldragontattoo if you have any questions. 


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Check the current browser support cutoff list](https://cfpb.github.io/consumerfinance.gov/browser-support#current-browser-support-metrics) for browsers that are advisable
to prioritize for testing.

#### Accessibility


#### Other
